pkg/provider/vault/vault.go

```diff
@@ -695,29 +695,35 @@ func (v *client) buildMetadataPath(path string) (string, error) {
 }
 func (v *client) buildPath(path string) string {
 	optionalMount := v.store.Path
-	origPath := strings.Split(path, "/")
-	newPath := make([]string, 0)
-	cursor := 0
-
-	if optionalMount != nil && origPath[0] != *optionalMount {
-		// Default case before path was optional
-		// Ensure that the requested path includes the SecretStores paths as prefix
-		newPath = append(newPath, *optionalMount)
-	} else {
-		newPath = append(newPath, origPath[cursor])
-		cursor++
-	}
-
-	if v.store.Version == esv1beta1.VaultKVStoreV2 {
-		// Add the required `data` part of the URL for the v2 API
-		if len(origPath) < 2 || origPath[1] != "data" {
-			newPath = append(newPath, "data")
+	out := path
+	// if optionalMount is Set, remove it from path if its there
+	if optionalMount != nil {
+		cut := *optionalMount + "/"
+		if strings.HasPrefix(out, cut) {
```

Comment:
This if statement relates to an existing bug if users have paths like `mount/path/data/mount/path/secret` (they would be never reachable), but keeps the code without any breaking changes. If we remove it, the behavior will change from:
```
path: foo/bar
key: foo/bar[/data]/secret
resulting path: /foo/bar[/data]/secret
```
into:
```
path: foo/bar
key: foo/bar[/data]/secret
resulting path: /foo/bar[/data]/foo/bar[/data]/secret
```
[/data] depends whether the kv version is V2 or V1